lib/stream.js

```diff
@@ -112,16 +117,14 @@ export const getSpawnedResult = async (
 	cleanupOnExit(spawned, cleanup, detached, finalizers);
 	const customStreams = getCustomStreams(stdioStreams);
 
-	const stdoutPromise = getStreamPromise(spawned.stdout, {encoding, buffer, maxBuffer});
-	const stderrPromise = getStreamPromise(spawned.stderr, {encoding, buffer, maxBuffer});
+	const stdioPromises = spawned.stdio.map((stream, index) => getStdioPromise({stream, index, stdioStreams, encoding, buffer, maxBuffer}));
 	const allPromise = getStreamPromise(spawned.all, {encoding, buffer, maxBuffer: maxBuffer * 2});
 
 	try {
 		return await Promise.race([
 			Promise.all([
 				once(spawned, 'exit'),
-				stdoutPromise,
-				stderrPromise,
+				Promise.all(stdioPromises),
```

Comment:
```suggestion
				...stdioPromises,
```

?

Reply:
The problem is that there are two variadic arguments: `stdioPromises` and `customStreams`, i.e. the length of those arrays is variable. But we still want to retrieve the resolved values of `stdioPromises` (not `customStreams`). 

I.e. we are doing this:

```js
const [
  [exitCode, signal, error],
  stdioResults,
  allResult,
] = Promise.all([
  once(spawned, 'exit'),
  Promise.all(stdioPromises), // length of `stdioPromises` is variable
  allPromise,
  ...customStreams.map(customStream => ...), // length of `customStreams` is variable
])
```

If we do this instead:

```js
const [
  [exitCode, signal, error],
  allResult,
  ...stdioResults,
] = Promise.all([
  once(spawned, 'exit'),
  allPromise,
  ...stdioPromises,
  ...customStreams.map(customStream => ...), 
])
```

`stdioResults` will get the results from both `stdioPromises` and `customStreams`.

Reply:
You are right. I didn't consider that.

Reply:
I think it's because it is a little hard to see what's going on in there.

But I believe when we were previously passing promises around, we were obfuscating that complexity. Having a single function that shows the full flow with `async`/`await` is a little easier to wrap our head around, although it's still inherently complex.